src/lib/fragments/ComponentTemplate.react.js

```diff
@@ -0,0 +1,18 @@
+import React, { Component } from 'react';
+
+import { defaultProps, propTypes } from '../components/ComponentTemplate.react';
+
+class ComponentTemplate extends Component {
```

Comment:
Can we use functional components instead?

```
const ComponentTemplate = (props) =>{
    const {
        id
    } =  { ...defaultProps,...props };

    return (
        <div id={id}>
            I love React.js!
        </div>
    ) 
}
```

Reply:
I was, but the [React guidance from Plotly use class based](https://dash.plotly.com/react-for-python-developers#our-very-own-react-component), and I want to lean on their content as much as possible for the session.

I also wonder if moving the functional components to class based would make it easier for DLUHC too? 🤔

Reply:
Shame about plotly using class components in their examples, as all our components in that repo use functional components. 

I wouldn't look at changing back to class components, mainly for the AutoComplete component as it uses react hooks for managing internal state for large parts of the functionality. This is what we had issues with the past couple of days, and would be a nightmare to rewrite. The others are simpler but we'd end up with a mixture of class and functional which would be confusing.

My preference would be do everything in functional so DLUHC can use our current components as examples, but that plotly documentation page does have a lot of content using classes 🤔

Reply:
Yeah, I think for now I'd lean towards having the template functional as the other components are functional and it will help give a simple example of how components are set up when working on the existing components. 

With the amount of hooks being used, it would take a bit of work to change the others to be classes so unless it's something we wanted to update very soon it might make it a bit more confusing to have a mix.

Reply:
We briefly spoke about the trade-offs here:

1. Existing documentation provided by Plotly uses class based components.
2. Our code uses functional components
3. The autocomplete component was class based, but is now functional.  Migrating back again might be hard? Hard to tell without trying.  It's also hard to identify the specific changes that Jordan made to the component.
4. We thought that learning general with class based components and then switching over to functional components once you've got the hand of React in general wouldn't be too hard.
5. There isn't much time between now and the learning session, so will leave the example as class based for now, and add some content around class based vs functional components.